fix: update API reference for set_integer_now_func. (#3135)
timescale/docs: api/set_integer_now_func.md

  [chunk pruning](https://www.timescale.com/blog/optimizing-queries-timescaledb-hypertables-with-partitions-postgresql-6366873a995d/) is executed at runtime. This generates a correct result, but may increase 
  planning time.

`set_integer_now_func` does not work on tables where the `time` column type is `TIMESTAMP`, `TIMESTAMPTZ`, or 
`DATE`.  

## Required arguments

|Name|Type| Description |
|-|-|-|
|`main_table`|REGCLASS| The hypertable `integer_now_func` is used in. |
|`integer_now_func`|REGPROC| A function that returns the current time set in each row in the `time` column in `main_table`.|

## Optional arguments

|Name|Type| Description|
|-|-|-|
|`replace_if_exists`|BOOLEAN| Set to `true` to override `integer_now_func` when you have previously set a custom function. Default is `false`. |

## Sample usage

Set the integer `now` function for a hypertable with a time column in [unix time](https://en.wikipedia.org/wiki/Unix_time).

- `IMMUTABLE`: when you execute the query each time: 
    ```sql
    CREATE OR REPLACE FUNCTION unix_now_immutable() returns BIGINT LANGUAGE SQL IMMUTABLE as $$  SELECT extract (epoch from now())::BIGINT $$;
    

>     SELECT set_integer_now_func('test_table_bigint', 'unix_now_immutable');

Reviewer: ```suggestion
    SELECT set_integer_now_func('hypertable_name', 'unix_now_immutable');
```

    ```

- `STABLE`: for prepared statements:
    ```sql
    CREATE OR REPLACE FUNCTION unix_now_stable() returns BIGINT LANGUAGE SQL STABLE AS $$ SELECT extract(epoch from now())::BIGINT $$;
    

>     SELECT set_integer_now_func('test_table_bigint', 'unix_now_stable');

Reviewer: Can we also add the hypertable that will use it? Just because it says `test_table_ bigint` Or:

```suggestion
    SELECT set_integer_now_func('hypertable_name', 'unix_now_stable');
```

# set_integer_now_func()

Override the [`now()`](https://www.postgresql.org/docs/16/functions-datetime.html) date/time function used to
set the current time in the integer `time` column in a hypertable. Many policies only apply to 

> [chunks][hash-partitions] of a certain age. `integer_now_func` determines the age of each chunk.

Reviewer 2: Not sure why you call this "hash partitioning" since it is linking to general partitioning functions. It's only the link name, but so it is not visible to users, but it is a little confusing when you you want to edit the documentation.

Author: I copied and pasted the link from another page :-). 

Updated to match the link text.

  planning time.

`set_integer_now_func` does not work on tables where the `time` column type is `TIMESTAMP`, `TIMESTAMPTZ`, or 
`DATE`.  

## Required arguments

|Name|Type| Description |
|-|-|-|
|`main_table`|REGCLASS| The hypertable `integer_now_func` is used in. |
|`integer_now_func`|REGPROC| A function that returns the current time set in each row in the `time` column in `main_table`.|

## Optional arguments

|Name|Type| Description|
|-|-|-|
|`replace_if_exists`|BOOLEAN| Set to `true` to override `integer_now_func` when you have previously set a custom function. Default is `false`. |

## Sample usage

Set the integer `now` function for a hypertable with a time column in [unix time](https://en.wikipedia.org/wiki/Unix_time).

- `IMMUTABLE`: when you execute the query each time: 
    ```sql
    CREATE OR REPLACE FUNCTION unix_now_immutable() returns BIGINT LANGUAGE SQL IMMUTABLE as $$ SELECT extract(epoch 
  from now())::BIGINT $$;
    
    SELECT set_integer_now_func('test_table_bigint', 'unix_now_immutable');
    ```

- `STABLE`: for prepared statements:
    ```sql

>     CREATE OR REPLACE FUNCTION unix_now_stable() returns BIGINT: Can you supply an example please. 

Reviewer 2: Maybe be explicit with the `STABLE` even though it is the default. You can use the same code as above.

```suggestion
    CREATE OR REPLACE FUNCTION unix_now_stable() returns BIGINT LANGUAGE SQL STABLE AS $$ SELECT extract(epoch from now())::BIGINT $$; 
```
